### PR TITLE
HPCC-28069 Default logging to non-dropping mode in K8s.

### PR DESCRIPTION
**Pull request by jakesmith**
To match bare-metal behaviour, logs should not by default drop
logs when queue hits the limit (default behaviour is to block).
In bare-metal - there are couple of exceptions, roxie and
the topology server configure logging to drop logs when queue
is full rather than block.
Preserve that behaviour via default configuration.

Signed-off-by: Jake Smith <jake.smith@lexisnexisrisk.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

File: helm/hpcc/values.schema.json [hunk 1]
```diff
@@ -928,6 +928,16 @@
         "detail": {
           "type": "integer",
           "description": "Log output verbosity"
+        },
```
Comment (rpastrana) [inline helm/hpcc/values.schema.json:931]: Might be helpful to document these new options in the README, and PDF
[queueDropDefault ](https://github.com/hpcc-systems/HPCC-Platform/tree/master/helm/examples/logging#hpcc-systems-application-level-logging-details)

File: system/jlib/jlog.cpp [hunk 2]
```diff
@@ -2453,14 +2446,21 @@ void setupContainerizedLogMsgHandler()
...
-            unsigned queueLen = logConfig->getPropInt(logQueueLenAtt, queueLenDefault);
-            unsigned queueDrop = logConfig->getPropInt(logQueueDropAtt, queueDropDefault);
-
             queryLogMsgManager()->enterQueueingMode();
-            queryLogMsgManager()->setQueueDroppingLimit(queueLen, queueDrop);
+            unsigned queueDrop = logConfig->getPropInt(logQueueDropAtt, queueDropDefault);
+            if (queueDrop)
+            {
+                queryLogMsgManager()->setQueueDroppingLimit(queueLen, queueDrop);
+                PROGLOG("JLog: queuing enabled with dropping: queueLen=%u, queueDrop=%u", queueLen, queueDrop);
```
Comment (rpastrana) [inline system/jlib/jlog.cpp:2457]: somewhat subjective and likely to change, but this is how PROLOG and INFOLOG have been defined:
    MSGCLS_information = 0x08, // Config, environmental and internal status  info
    MSGCLS_progress    = 0x10, // Progress of workunits. Status of file operations
https://github.com/hpcc-systems/HPCC-Platform/blob/a1240fda33e1110602607e5ab876ab4700b17238/system/jlib/jlog.hpp#L66

**Review by rpastrana — COMMENTED** [review]
@jakesmith see comments

File: roxie/roxie/roxie.cpp [hunk 3]
```diff
@@ -51,6 +51,9 @@ version: "1.0"
   localSlave: true
   numChannels: 1
   queueNames: roxie.roxie
+  logging:
+    detail: 50
```
Comment (ghalliday) [inline roxie/roxie/roxie.cpp:55]: Why 50?

**Review by ghalliday — COMMENTED** [review]
One question about detail level for roxie.

File: system/jlib/jlog.cpp (hunk 2 above)
Comment (jakesmith) [reply system/jlib/jlog.cpp:2457]: Fair enough, however, PROGLOG is used widely for this purpose, I'd say it is used more for 'internal status'/progress than it is for "Progress of workunits. Status of file operations", but as you say, all a bit subjective.

In fact, it looks like they're being used the opposite way around from what the comments pescribe.
MCuserInfo is being used in 22 places at the moment, and they're mostly related to workunit/files.
PROGLOG is used in 1,624 places, mostly for internal status info purposes.

Perhaps we should flip the comments in jlog.hpp at this stage? (not this PR)

File: helm/hpcc/values.schema.json (hunk 1 above)
Comment (jakesmith) [reply helm/hpcc/values.schema.json:931]: Will add to the readme along with the others, but they're currently in helm/examples/logging/README.md.
examples/logging makes sense for setting up the logging stack setup, but the "logging:" settings would be better placed in helm/hpcc/README.md I think, as they are pertinent with or without a logging stack configuration.

**Review by ghalliday — APPROVED** [review]

**Pull request merged**